src/Raven.Server/Documents/Indexes/Index.cs

```diff
@@ -3222,7 +3222,15 @@ private int MinBatchSize
 
                 if (Configuration.TransactionSizeLimit == null)
                 {
-                    _transactionSizeLimit = new Lazy<Size?>(() => null);
+                    _transactionSizeLimit = new Lazy<Size?>(() =>
+                    {
+                        if (DocumentDatabase.IsEncrypted)
```

Comment:
I think we need to enhance the logic here. Notice that we are doing some adjustments for map-reduce when the value is not null.

maybe sth like to simply the code here:
```
var limit = DocumentDatabase.IsEncrypted 
    ? Configuration.EncryptedTransactionSizeLimit ?? Configuration.TransactionSizeLimit 
    : Configuration.TransactionSizeLimit;

if (limit != null && Type.IsMapReduce())
    limit = limit * 0.75;

_transactionSizeLimit = new Lazy<Size?>(() => limit);
```